### ISSUE TITLE
`react-native-reanimated` expo plugin broken after introduction of `@rnx-kit/react-native-host`

### ISSUE THREAD

**Issue by rrebase** [microsoft/react-native-test-app#1428]
### What happened?

The bundled [reanimated plugin](https://github.com/microsoft/react-native-test-app/wiki/Config-Plugins#bundled-plugins) stopped working after https://github.com/microsoft/react-native-test-app/commit/241ddaf83e74f5d53a9144bcfd52f8f948425fcb due to removal of `ReactTestApp/BridgeDelegate.mm`, which is now located inside the `@rnx-kit/react-native-host` package.

The relevant expo plugin provider is setup [here](https://github.com/microsoft/react-native-test-app/blob/trunk/scripts/config-plugins/plugins/withIosBaseMods.mjs#L33-L36). 

`makeFilePathModifier` calls `findFile` and in case of invalid path, it goes to an infinite loop. Thus as a side-note, I propose to cap it at X steps and then throw error about file not being found.

### Version

`>=2.4.0`

### What platforms are you seeing this issue on?

- [X] Android (as gradle tries to apply plugins if expo plugins are installed and get stuck trying to find a file that doesn't exist)
- [X] iOS

### Steps to Reproduce

```diff
diff --git a/example/App.js b/example/App.js
index c9cd14d..b3bce70 100644
--- a/example/App.js
+++ b/example/App.js
@@ -10,6 +10,7 @@ import {
   useColorScheme,
   View,
 } from "react-native";
+import Animated from "react-native-reanimated";
 import { SafeAreaProvider, SafeAreaView } from "react-native-safe-area-context";
 // @ts-expect-error
 import { version as coreVersion } from "react-native/Libraries/Core/ReactNativeVersion";
@@ -179,6 +180,16 @@ const App = ({ concurrentRoot }) => {
         >
           <Header />
           <DevMenu />
+          <Animated.View
+            style={[
+              {
+                width: 100,
+                height: 80,
+                backgroundColor: "orange",
+                margin: 30,
+              },
+            ]}
+          />
           <View style={styles.group}>
             <Feature value={getReactNativeVersion()}>React Native</Feature>
             <Separator />
diff --git a/example/app.json b/example/app.json
index ef69fa7..c492b30 100644
--- a/example/app.json
+++ b/example/app.json
@@ -13,6 +13,9 @@
       "presentationStyle": "modal"
     }
   ],
+  "plugins": [
+    "react-native-test-app/plugins/reanimated.js"
+  ],
   "resources": {
     "android": [
       "dist/res",
diff --git a/example/babel.config.js b/example/babel.config.js
index 5ae49c3..8a4543d 100644
--- a/example/babel.config.js
+++ b/example/babel.config.js
@@ -1,3 +1,4 @@
 module.exports = {
   presets: ["module:metro-react-native-babel-preset"],
+  plugins: ["react-native-reanimated/plugin"],
 };
diff --git a/example/package.json b/example/package.json
index a51d4cf..e2230e2 100644
--- a/example/package.json
+++ b/example/package.json
@@ -24,6 +24,7 @@
   },
   "devDependencies": {
     "@babel/core": "^7.0.0",
+    "@expo/config-plugins": "^5.0.0",
     "@types/jest": "^27.0.0",
     "@types/react-native": "^0.68.0",
     "jest": "^27.0.0",
@@ -32,6 +33,7 @@
     "react": "17.0.2",
     "react-native": "^0.68.2",
     "react-native-macos": "^0.68.3",
+    "react-native-reanimated": "^2.0",
     "react-native-safe-area-context": "^4.3.4",
     "react-native-test-app": "workspace:.",
     "react-native-windows": "^0.68.8"
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by rrebase** [comment]
Put up a draft PR to update the provider to reference relevant file from `@rnx-kit/react-native-host` (https://github.com/microsoft/react-native-test-app/pull/1429), but found that more than just the file location changed as there's an error trying to resolve a header in reanimated file. Didn't dive into the details there, assistance in updating that part would be appreciated if the approach makes sense 🙏

**Comment by tido64** [comment]
Thanks for filing and working on this. I've got a fix for the infinite loop. Will take a look at the other issue.

**Comment by tido64** [comment]
I think I've got it all figured out. We need to make a small change in `@rnx-kit/react-native-host` to make one of the header paths available to us. The plugin also needed to define the `REACT_NATIVE_MINOR_VERSION` macro. We didn't need this before, which to me is a bit of a "how did this ever work" moment.

**Comment by tido64** [comment]
This should now be fixed in [2.5.4](https://github.com/microsoft/react-native-test-app/releases/tag/2.5.4). Let us know if you still have issues after bumping.

Thanks again for doing the initial investigation. Really helped us with getting the fix out quickly.
